### PR TITLE
[MPDX-9870] Complete status chip when required goal fields are filled

### PR DESCRIPTION
**Pull request by canac**
## Description

- The Goal Settings (Admin) header status chip was hard-coded to an amber "Incomplete" and never changed.
- It now shows a green "Complete" chip once the required fields are filled, and "Incomplete" otherwise.
- Required fields: calculation year, benefits plan, and — for each person on the calculation (both spouses when married) — age, full-time years on staff, and role (Field/Office).
- Added `isGoalSettingsComplete(values)` helper (pure, unit-tested) driven by the live Formik values, so the chip updates as fields are edited.

Jira: MPDX-9870

## Testing

- Go to a coaching account's New Staff Goal Calculator, Admin **Goal Settings** view (`?view=goal-settings`) (e.g. `/accountLists/6bbc265f-ec81-4b23-b297-5bd298c4d619/coaching/90935daa-0e49-4b31-a41b-d6b46854ebca/nsGoalCalculator?view=goal-settings`).
- With a required field empty (e.g. **Benefits Selection**), confirm the header chip reads **Incomplete** (amber).
- Fill in the remaining required fields — calculation year, age + full-time years on staff (both spouses if married), Field/Office, and Benefits Selection.
- Check that the chip flips to a green **Complete** chip once the last required field is set **before saving**.
- For a married couple, confirm the chip stays **Incomplete** until both spouses' age and years-on-staff are filled.

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels (_Add the label "Preview" to automatically create a preview environment_)
- [x] I have run the Claude Code `/pr-review` command locally and fixed any relevant suggestions
- [ ] I have requested a review from another person on the project
- [x] I have tested my changes in preview or in staging
- [x] I have cleaned up my commit history

### CODE REVIEW

File: src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * Whether the Goal Settings form has the fields required to produce a valid
+ * goal. Age, full-time years on staff, and role (field/office) are required for
+ * each person on the calculation — both spouses when married — and a
+ * calculation year and benefits plan are required for the household. Drives the
+ * header's Complete/Incomplete status chip.
+ */
+export const isGoalSettingsComplete = (
+  values: GoalSettingsFormValues,
+): boolean => {
+  const isMarried =
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts:14]: [Suggestion] (Architecture, 4.0/10) The married check `maritalStatus === Married` now lives in three places in this folder: here, `GoalSettingsForm.tsx:134` (`hasSpouse`), and `goalSettingsApiMapping.ts:122` (`hasSpouse`). The completion predicate's correctness is coupled to the API mapping's `hasSpouse` gate (a spouse field is only "required" if it's also going to be sent), so these should share one source of truth. Consider extracting a small `hasSpouse(values)` helper next to `GoalSettingsFormValues` and reusing it here. Non-blocking — the duplication predates this PR.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    values.maritalStatus === NewStaffQuestionnaireMaritalStatusEnum.Married;
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts:15]: [Medium] (Testing, 5.0/10) `maritalStatus === ''` evaluates to "not married," so a household with the 5 single-required fields filled but marital status still unset renders a green "Complete" chip. `maritalStatus` maps from a null API value to `''` on load, so this state is reachable, and it isn't itself in `requiredFields`. Whether an unknown marital status should gate completeness is a product call — but it's currently untested either way. Add a test documenting the intended behavior (and, if unset marital status should block, add it to the required set).

File: src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  const requiredFields: Array<string | number | null | undefined> = [
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.ts]: [Suggestion] (Financial Reporting, 4.0/10) The required set intentionally omits goal-affecting money inputs (`annualRequestedSalary`, `contribution403bPercentage`, `ministryExpenses`, senior-staff transfers). This is defensible — the calc-year tooltip says salary/benefit constants are derived from age + tenure + role + year, and 403(b) is prefilled — but I couldn't confirm from client code that a blank `annualRequestedSalary` yields a non-zero server `monthlyGoal`. A reviewer with backend knowledge should confirm salary is truly derived (not required) so a green "Complete" chip never sits atop a goal computed from a $0 salary.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsForm.tsx [hunk 4]
```diff
@@ -166,6 +167,7 @@ export const GoalSettingsForm: React.FC<GoalSettingsFormProps> = (props) => {
                 mpdGoal={mpdGoal}
                 joinedStaffYear={calculation.joinedStaffYear}
                 isScenario={isScenario}
+                isComplete={isGoalSettingsComplete(values)}
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsForm.tsx:170]: [Medium] (UX, 5.0/10) `isGoalSettingsComplete(values)` and the Save button's `disabled={!isValid || isSubmitting}` are independent signals. The Yup schema marks every field `.nullable()`/optional, so the chip can show "Complete" while Save is disabled by an invalid optional field (e.g. a 403(b) % > 99). "Complete" implies "ready to save," so the two disagreeing is mildly confusing. Consider whether "Complete" should also require `isValid`, or document that the chip reflects goal-field completeness only, not save-readiness.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsHeader.tsx [hunk 5]
```diff
@@ -126,6 +131,13 @@ export const GoalSettingsHeader: React.FC<GoalSettingsHeaderProps> = ({
             label={t('Scenario Only')}
             size="small"
           />
+        ) : isComplete ? (
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsHeader.tsx:134]: [Suggestion] (UX, 4.0/10) The chip text flips Incomplete→Complete live as the user fills the last required field, but it's plain non-interactive text with no `aria-live` region — a screen-reader user editing the form isn't notified of the status change. Matches the pre-existing Incomplete/Scenario chips, so this is a nice-to-have: wrap the chip in a container with `role="status"` / `aria-live="polite"` if you want the transition announced. Color-as-sole-indicator is fine here (both chips carry a text label).

File: src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.test.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  assignmentType: GoalCalculationRole.Field,
+  benefitsPlan: MpdGoalBenefitsConstantPlanEnum.Base,
+};
+
+describe('isGoalSettingsComplete', () => {
+  it('is false when required fields are empty', () => {
+    expect(isGoalSettingsComplete(emptyValues)).toBe(false);
+  });
+
+  it('is true when a single staff has age, tenure, role, and benefits plan', () => {
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/goalSettingsCompletion.test.ts:68]: [Suggestion] (Testing, 3.5/10) Completeness ignoring the ~20 optional financial fields is only implicitly proven (`filledSingle` leaves them blank and returns `true`). Add an intent-naming test so a future change that accidentally widens `requiredFields` fails loudly:
```ts
it('is complete even when optional financial fields are blank', () => {
  expect(isGoalSettingsComplete({ ...filledSingle, annualRequestedSalary: '', ministryExpenses: '' })).toBe(true);
});
```

File: src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsForm.test.tsx [hunk 7]
```diff
@@ -445,6 +450,85 @@ describe('GoalSettingsForm', () => {
...
+          goalCalculationMock={{
+            newStaffGoalCalculation: {
+              ...completeMock.newStaffGoalCalculation,
+              spouseAge: null,
+            },
+          }}
+        />,
+      );
+
+      expect(await findByText('Incomplete')).toBeInTheDocument();
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsForm.test.tsx:506]: [Suggestion] (Testing, 3.0/10) Asymmetric with its sibling tests at :475 and :491 — this "spouse required fields missing" case asserts `findByText('Incomplete')` but doesn't also assert `queryByText('Complete')` is absent. Add the negative assertion for parity.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsHeader.test.tsx [hunk 8]
```diff
@@ -82,10 +85,18 @@ describe('GoalSettingsHeader', () => {
...
-  it('marks the calculation as incomplete', () => {
-    const { getByText } = render(<TestComponent />);
+  it('marks the calculation as incomplete when required fields are unfilled', () => {
+    const { getByText, queryByText } = render(<TestComponent />);
 
     expect(getByText('Incomplete')).toBeInTheDocument();
+    expect(queryByText('Complete')).not.toBeInTheDocument();
+  });
+
+  it('marks the calculation as complete when required fields are filled', () => {
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/GoalSettingsHeader.test.tsx:95]: [Suggestion] (Testing, 3.0/10) The chip ternary is `isScenario` > `isComplete` > incomplete, but no test renders `<TestComponent isScenario isComplete />` to assert "Scenario Only" still wins (no "Complete" chip). Structurally guaranteed but cheap to pin down against a future reorder.

**Review by canac — COMMENTED** [review]
## Multi-Agent Code Review — PR #1934

**Verdict: APPROVED WITH SUGGESTIONS** · Risk **3/10 LOW** · **0 blockers**

5 specialized agents (Architecture, Testing, Standards, UX, Financial Reporting) + a dependency-impact pass reviewed this PR. Clean, well-tested, self-contained change. Standards fully compliant; test suite runs 62/62 green with ESLint clean. No critical, high, or important findings — every item below is a non-blocking suggestion.

### Highlights
- **Required-field set is correct.** The Financial agent investigated the suspected `spouseJoining` bug and refuted it: spouse age/tenure are rendered *and* sent to the API whenever `maritalStatus === Married` (independent of joining status), so keying the predicate off `Married` exactly matches both the UI and `formValuesToAttributes`. Keying off `spouseJoining` would have been the bug.
- **Chip matches an existing convention** — byte-for-byte identical to the status chip in `GoalsTable.tsx` (same `size`/`variant`/`color`/labels).
- **Dependency impact: LOW** — `isComplete` is optional with a `false` default, so no existing caller breaks; all 8 predicate fields verified present on `GoalSettingsFormValues`.

### Suggestions (severity 5.0)
- **`GoalSettingsForm.tsx:170`** — "Complete" chip and Save-enabled (`!isValid`) are independent signals; the chip can read "Complete" while Save is disabled by an invalid optional field.
- **`goalSettingsCompletion.ts:18`** — `maritalStatus === ''` is treated as "not married," so an unset marital status (reachable on load) can render green "Complete".

### Minor suggestions (<5.0)
- Married check now duplicated 3× — extract a shared `hasSpouse(values)`.
- Live chip flip isn't in an `aria-live`/`role="status"` region.
- "Complete" while salary/expenses blank rests on an unverified server salary-derivation assumption — a reviewer with backend knowledge should confirm a blank salary can't yield a $0 goal.
- Test polish: add an intent test that optional financial blanks don't block completeness; add the missing negative `queryByText('Complete')` in the spouse-missing test; pin down scenario-precedence-over-complete.

_Pre-existing (informational, not from this PR): the amber "Incomplete" chip's orange text is borderline contrast for small text._

Severity < 7 findings can be dismissed by replying `/dismiss: <reason>` on the specific comment.
